File: benchmark/results/java-invocation-script/run-001/b-baseline/output/com/metaengine/demo/catalog/services/ProductRepository.java
package com.metaengine.demo.catalog.services;

import com.metaengine.demo.catalog.aggregates.Product;
import java.util.List;

/** ProductRepository for the catalog domain. */
public class ProductRepository {

    /** Persist a new Product. */
    public Product create(Product input) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Load a Product by id; may return null. */
    public Product findById(String id) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** List products up to the given limit. */
    public List<Product> list(int limit) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Delete a product by id. */
    public void delete(String id) {
        throw new UnsupportedOperationException("not implemented");
    }
}

File: benchmark/results/python-multilang/run-002/b-baseline/output/metaengine_demo/catalog/services/product_repository.py
"""ProductRepository for the catalog domain."""

from typing import Optional

from metaengine_demo.catalog.aggregates.product import Product


class ProductRepository:
    """Persistence boundary for the Product aggregate."""

    def create(self, input: Optional[Product]) -> Product:
        """Persist a new product from a partial input."""
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[Product]:
        """Load a product by its identifier."""
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[Product]:
        """List persisted products up to the given limit."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Delete a persisted product by identifier."""
        raise NotImplementedError("not implemented")
